### ISSUE TITLE
PHP 8.4 support

### ISSUE THREAD

**Issue by tobimori** [tobimori/kirby-dreamform#100]

**Comment by samlinck** [comment]
When trying to use the Kirby CLI commands Dreamform throws an error:

`Whoops\Exception\ErrorException: tobimori\DreamForm\Actions\Action::log(): Implicitly marking parameter $type as nullable is deprecated, the explicit nullable type must be used instead in /site/plugins/kirby-dreamform/classes/Actions/Action.php:70`

[Info](https://php.watch/versions/8.4/implicitly-marking-parameter-type-nullable-deprecated)

**Comment by tobimori** [comment]
please install the current develop branch for Kirby 5 and PHP 8.4 support.

**Comment by WebMechanic** [comment]
Hey Tobi,

we still require Kirby 4 compatability and this issue kinda-sorta prevents us from using 8.4. It appears that turning off `E_DEPRECATION` doesn't do the job (which I find a bit odd).

Is there an intermediate release in the 1.5.x RCs that includes the fix for "explicit nullable type" we could use meanwhile or was this only addressed for the 2.0 branch and K5? 

I found a few commits where method signatures were updated but wouldn't want to pick a random commit from February/March this year and call it a day... :)

Thank you.

**Comment by tobimori** [comment]
Yes, the 1.5.x RC branch was abandoned (not all method signatures have been updated in that one) because I was stuck on an issue with Kirby Core that was only fixed in 5.0.

I'm not sure what you've tried, I was able to put:

```error_reporting(E_ALL & ~E_DEPRECATED & ~E_USER_DEPRECATED);```
 
at the top of my index.php and that solved the issue.

**Comment by WebMechanic** [comment]
thanks, I completely forgot about `E_USER_DEPRECATED`
however .... with error_reporting() _**not working either**_ I dug deeper, made a full repo text search and found a stray `error_reporting(E_ALL)` in one of our own, very early plugin files... 🙄

added `E_ALL & ~E_DEPRECATED & ~E_USER_DEPRECATED` to php.ini and it's "solved".
php 8.4 update went well.

Danke für den mentalen Schubser 😁

Cheers!
